docs/content/topics/operators/gateways.md

```diff
@@ -11,340 +11,518 @@ aliases:
+  "github.com/brigadecore/brigade/sdk/v2/authn"
+  "github.com/brigadecore/brigade/sdk/v2/restmachinery"
 )
 
-cat <<EOF | kubectl --namespace ${namespace} create -f -
-apiVersion: v1
-kind: Secret
-metadata:
-  name: ${name}
-  labels:
-    heritage: brigade
-    project: ${project_id}
-    build: ${uuid}
-    component: build
-type: "brigade.sh/build"
-data:
-  revision:
-    commit: $(base64 -w 0 <<<"${commit_id}")
-    ref: $(base64 -w 0 <<<"${commit_ref}")
-  event_provider: $(base64 -w 0 <<<"${event_provider}")
-  event_type: $(base64 -w 0 <<<"${event_type}")
-  project_id: $(base64 -w 0 <<<"${project_id}")
-  build_id: $(base64 -w 0 <<<"${uuid}")
-  payload: $(base64 -w 0 <<<"${payload}")
-  script: $(base64 -w 0 <<<"${script}")
-EOF
+func main() {
+  ctx := context.Background()
+
+  // Get the Brigade API server address and root user password from env
+  apiServerAddress, err := brigadeOS.GetRequiredEnvVar("APISERVER_ADDRESS")
+  if err != nil {
+    log.Fatal(err)
+  }
+  rootPassword, err := brigadeOS.GetRequiredEnvVar("APISERVER_ROOT_PASSWORD")
+  if err != nil {
+    log.Fatal(err)
+  }
+
+  // Create a root session to acquire an auth token
+
+  // The default Brigade deployment mode uses self-signed certs
+  // Hence, we allow insecure connections in our APIClientOptions
+  apiClientOpts := &restmachinery.APIClientOptions{
+    AllowInsecureConnections: true,
+  }
+
+  // Create a new auth client
+  authClient := authn.NewSessionsClient(
+    apiServerAddress,
+    "",
+    apiClientOpts,
+  )
+
+  // Create a root session, which returns an auth token
+  token, err := authClient.CreateRootSession(ctx, rootPassword)
+  if err != nil {
+    log.Fatal(err)
+  }
+  tokenStr := token.Value
+}
 ```
 
-Next, we will package the above as a Docker image. To do that, we create a `Dockerfile`
-in the same directory as the `cron-event.sh` script above.
-
-The `Dockerfile` just sets up the commands we need, and then copies the script into
-the image:
+We now have the code needed for the example gateway to communicate with the
+Brigade API Server and acquire an authentication token via a root user session.
+
+### Event Creation
+
+Let's add logic to the gateway so that it can create an event for a provided
+Brigade project. We'll focus first on the additions needed for creating an
+event in a new function, `createEvent`, that we'll link together with the
+`main` function afterwards.
+
+```go
+// createEvent creates a Brigade Event for the provided project, using the
+// provided sdk.Client
+func createEvent(client sdk.APIClient, projectID string) error {
+  ctx := context.Background()
+
+  // Construct a Brigade Event
+  event := core.Event{
+    // This is the ID/name of the project that the event will be intended for
+    ProjectID: projectID,
+    // This is the source value for this event
+    Source:    "brigade.sh/example-gateway",
+    // This is the event's type
+    Type:      "create-event",
```

Comment:
With the payload being "Dolly," you don't want to make this a "hello" event?? 😉

Reply:
Yes!  Good call.